### PR TITLE
chore: consolidate workflows into understack_workflows

### PR DESCRIPTION
**Pull request by skrobul**
This PR moves:
- `nautobot-update-cf`
- `ironic-to-nautobot-sync`
- `undersync-device`

workflows into the shared `ironic-nautobot-client` container image/python package.

Closes https://rackspace.atlassian.net/browse/PUC-421

### CODE REVIEW

File: python/understack-workflows/understack_workflows/main/nautobot_update_cf.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+from understack_workflows.helpers import setup_logger
+from understack_workflows.nautobot import Nautobot
+
+
+def argument_parser():
+    parser = argparse.ArgumentParser(
+        prog=os.path.basename(__file__),
+        description="Ironic to Nautobot provisioning state sync",
+    )
+    parser.add_argument("--device_uuid", required=True, help="Nautobot device UUID")
```
Comment (cardoe) [inline python/understack-workflows/understack_workflows/main/nautobot_update_cf.py:16]: We can do this as a follow up but I think we should probably standardize this arg. The other entry points use `--device-id` and they use `type=UUID` to validate the input

File: python/understack-workflows/understack_workflows/main/undersync_device.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        logger.error(error)
+        sys.exit(2)
+
+
+def argument_parser():
+    parser = argparse.ArgumentParser(
+        prog=os.path.basename(__file__),
+        description="Trigger undersync run for a device",
+    )
+    parser.add_argument("--device_uuid", required=True, help="Nautobot device UUID")
```
Comment (cardoe) [inline python/understack-workflows/understack_workflows/main/undersync_device.py:51]: We can do this as a follow up but I think we should probably standardize this arg. The other entry points use `--device-id` and they use `type=UUID` to validate the input

File: python/understack-workflows/understack_workflows/main/undersync_device.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    parser.add_argument("--nautobot_url", required=False)
+    parser.add_argument("--nautobot_token", required=False)
+    parser.add_argument(
+        "--force",
+        type=boolean_args,
+        help="Call Undersync's force endpoint",
+        required=False,
+    )
+    parser.add_argument(
+        "--dry-run",
```
Comment (cardoe) [inline python/understack-workflows/understack_workflows/main/undersync_device.py:62]: How do you expect this to be called?

File: python/understack-workflows/understack_workflows/main/undersync_device.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    parser = argparse.ArgumentParser(
+        prog=os.path.basename(__file__),
+        description="Trigger undersync run for a device",
+    )
+    parser.add_argument("--device_uuid", required=True, help="Nautobot device UUID")
+    parser.add_argument("--network-name", required=True)
+    parser.add_argument("--nautobot_url", required=False)
+    parser.add_argument("--nautobot_token", required=False)
+    parser.add_argument(
+        "--force",
```
Comment (cardoe) [inline python/understack-workflows/understack_workflows/main/undersync_device.py:56]: How do you expect this to be called?

File: python/understack-workflows/understack_workflows/nautobot.py [hunk 5]
```diff
@@ -102,3 +103,38 @@ def bulk_create_interfaces(
                 self.logger.info(f"{interface.name} successfully created")
 
             return req
+
+    def device_by_id(self, device_id: str) -> NautobotDevice:
```
Comment (cardoe) [inline python/understack-workflows/understack_workflows/nautobot.py:107]:
```suggestion
    def device_by_id(self, device_id: UUID) -> NautobotDevice:
```

File: python/understack-workflows/understack_workflows/nautobot.py [hunk 6]
```diff
@@ -102,3 +103,38 @@ def bulk_create_interfaces(
...
+        device = self.session.dcim.devices.get(device_id)
+        if not device:
+            self.exit_with_error(f"Device {device_id} not found in Nautobot")
```
Comment (cardoe) [inline python/understack-workflows/understack_workflows/nautobot.py:110]:
```suggestion
            self.exit_with_error(f"Device {device_id!s} not found in Nautobot")
```

File: python/understack-workflows/understack_workflows/nautobot.py [hunk 7]
```diff
@@ -102,3 +103,38 @@ def bulk_create_interfaces(
...
+        return device
+
+    def device_interfaces(self, device_id: str):
+        return self.session.dcim.interfaces.filter(device_id=device_id)
+
+    def update_cf(self, device_id, field_name: str, field_value: str):
```
Comment (cardoe) [inline python/understack-workflows/understack_workflows/nautobot.py:116]:
```suggestion
    def update_cf(self, device_id: UUID, field_name: str, field_value: str):
```

File: python/understack-workflows/understack_workflows/nautobot.py [hunk 8]
```diff
@@ -102,3 +103,38 @@ def bulk_create_interfaces(
...
+        device = self.device_by_id(device_id)
+        device.custom_fields[field_name] = field_value
+        response = device.save()
+        self.logger.info(f"save result: {response}")
+        return response
+
+    def uplink_switches(self, device_id) -> list[str]:
```
Comment (cardoe) [inline python/understack-workflows/understack_workflows/nautobot.py:123]:
```suggestion
    def uplink_switches(self, device_id: UUID) -> list[str]:
```

File: python/understack-workflows/understack_workflows/main/undersync_device.py [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+
+def argument_parser():
+    parser = argparse.ArgumentParser(
+        prog=os.path.basename(__file__),
+        description="Trigger undersync run for a device",
+    )
+    parser.add_argument("--device_uuid", required=True, help="Nautobot device UUID")
+    parser.add_argument("--network-name", required=True)
+    parser.add_argument("--nautobot_url", required=False)
```
Comment (cardoe) [inline python/understack-workflows/understack_workflows/main/undersync_device.py:53]:
```suggestion
    parser.add_argument("--nautobot_url", required=False, default="http://nautobot-default.nautobot.svc.cluster.local")
```

**Review by cardoe — COMMENTED** [review]
Well I rebased it so hopefully that doesn't mess ya up but left some comments on some clean ups.

File: python/understack-workflows/understack_workflows/main/undersync_device.py (hunk 4 above)
Comment (skrobul) [reply python/understack-workflows/understack_workflows/main/undersync_device.py:56]: not sure what do you mean? This is called with a string provided as a workflow parameter, for example `--force=false`

**Review by cardoe — APPROVED** [review]

**Pull request merged**